src/PackageDependencyViewExtension/PackageDependencyView.xaml.cs

```diff
@@ -0,0 +1,232 @@
+﻿using Dynamo.Graph.Workspaces;
+using Dynamo.Wpf.Extensions;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Windows.Controls;
+using System.Windows.Media;
+
+namespace Dynamo.PackageDependency
+{
+    /// <summary>
+    /// Interaction logic for PackageDependencyView.xaml
+    /// </summary>
+    public partial class PackageDependencyView : UserControl
+    {
+        private DependencyTable table = new DependencyTable();
+
+        private WorkspaceModel currentWorkspace;
+
+        /// <summary>
+        /// Event handler for workspaceAdded event
+        /// </summary>
+        /// <param name="obj">workspace model</param>
+        internal void OnWorkspaceChanged(IWorkspaceModel obj)
+        {
+            if (obj is WorkspaceModel)
+            {
+                DependencyRegen(obj as WorkspaceModel);
+                currentWorkspace = obj as WorkspaceModel;
+                currentWorkspace.PropertyChanged += OnWorkspacePropertyChanged;
```

Comment:
I think here we should unsubscribe the old workspace before subscribing the new one - even if they are usually the same.

Reply:
Added back